splink/case_statements.py

```diff
@@ -34,7 +34,7 @@ def _add_null_treatment_to_case_statement(case_statement: str):
 
     if "then -1" not in sl:
         try:
-            variable_name = re.search(r"when ([\w_]{1,100})_l", case_statement)[1]
+            variable_name = re.search(r"when ([\w_]{1,100})_l", case_statement.lower())[1]
```

Comment:
Not necessary for this PR, this was an error I identified whilst doing the PR that meant the regex was incorrect